packages/x-date-pickers/src/internals/hooks/useField/useField.utils.ts

```diff
@@ -387,64 +415,123 @@ export const splitFormatIntoSections = <TDate>(
   };
 
   for (let i = 0; i < expandedFormat.length; i += 1) {
+    const escapedPartOfCurrentChar = escapedParts.find(
+      (escapeIndex) => escapeIndex.start <= i && escapeIndex.end >= i,
+    );
+
     const char = expandedFormat[i];
 
-    if (char.match(/([A-zÀ-ú]+)/g)) {
+    if (escapedPartOfCurrentChar == null && char.match(/([A-Za-z]+)/g)) {
       currentTokenValue += char;
-    } else {
+    }
+    // If we are on the starting of the ending character of an escaped part of the format,
+    // Then we ignore this character.
+    else if (
+      escapedPartOfCurrentChar == null ||
+      (escapedPartOfCurrentChar.start !== i && escapedPartOfCurrentChar.end !== i)
+    ) {
       commitCurrentToken();
-      sections[sections.length - 1].separator += char;
+      if (sections.length === 0) {
+        startSeparator += char;
+      } else {
+        sections[sections.length - 1].separator += char;
+      }
     }
   }
 
   commitCurrentToken();
 
-  return sections.map((section) => {
+  const cleanSections = sections.map((section) => {
     if (section.separator !== '/') {
       if (section.separator !== null && section.separator.includes(' ')) {
         return {
           ...section,
           separator: `\u2069${section.separator}\u2066`,
-          parsingSeparator: section.separator,
         };
       }
       return section;
     }
     return {
       ...section,
       separator: ' / ',
-      parsingSeparator: '/',
     };
   });
+
+  return {
+    sections: cleanSections,
+    startSeparator,
+  };
 };
 
-export const createDateStrFromSections = (
+export const getDateFromDateStr = <TDate>(
+  utils: MuiPickersAdapter<TDate>,
+  dateStr: string,
+  format: string,
+) => {
+  const removeEscapedParts = (
+    stringToCleanup: string,
+    escapedPartsOfString: { start: number; end: number }[],
+  ) => {
+    let cleanedUpString = '';
+    for (let i = 0; i < stringToCleanup.length; i += 1) {
+      const char = stringToCleanup[i];
+      const isInEscapedPart = escapedPartsOfString.find(
+        (escapeIndex) => escapeIndex.start <= i && escapeIndex.end >= i,
+      );
+
+      if (!isInEscapedPart) {
+        cleanedUpString += char;
+      }
+    }
+
+    return cleanedUpString;
+  };
+
+  const expandedFormat = utils.expandFormat(format);
+  const escapedPartsInFormat = getEscapedPartsFromExpandedFormatOrDateStr(utils, expandedFormat);
+  const escapedPartsInDateStr = getEscapedPartsFromExpandedFormatOrDateStr(utils, dateStr);
```

Comment:
If I correctly understood the logic of `getEscapedPartsFromExpandedFormatOrDateStr` it's looking for escape token, why would the `dateStr` contain such token?

If I use `[The] 6th [of] December`, I expect the date string to be `The 6th of December` so not token to consider

Reply:
> If I use [The] 6th [of] December, I expect the date string to be The 6th of December so not token to consider

To be displayed in the input yes
To be parsed not
Because as explained in the PR description,  `dayjs` (and maybe others) don't handle parsing of formats with escaped characters. So before the parsing from the string, I need to remove the escaped characters from both the format and the stringified date.

Reply:
If the user past some text, it will be "The 6th of December", not the "[The] 6th [of] December" and we have 

```js
getEscapedPartsFromExpandedFormatOrDateStr("The 6th of December") -> []

getEscapedPartsFromExpandedFormatOrDateStr("[The] 6th [of] December") -> [{start: 0, end: 4}, {start: 10, end: 13}]
```

-----

I'm not sure that copy/past needs to support the escaped format, because you do not copy an element from an input to past it in the same input. data will probably come from external input for which we have no clue about the format used.

Would it make sense to use the default format for parsing the past text, and have a distinct way of customizing this format.

For example, I make an internal app, and intarnal tools sometime use `dd-mm` or `dd/mm/yyyy`, As a developper I would like to pass a prop to `dayJs` such that the past date is managed as follow:

```js
dayjs(inputValue, ["dd-mm", "dd/mm/yyyy"], 'fr', true);
```

Reply:
> If the user past some text, it will be "The 6th of December", not the "[The] 6th [of] December"

Right, I mixed up several things... 

____

> Would it make sense to use the default format for parsing the past text, and have a distinct way of customizing this format.

For the parsing, probably yes, to have something like Retool has where we test a list of formats, if possible first the format used in the field and then commonly used formats.
But that would be for later.

However, the logic we are talking about here also impacts imperative update of the field, to enable users to test there application with `fireEvents` without caring about the implementation detail of the field, just by firing an `onChange` with the new value.
And for this use case, they will probably match the format expected in the field.

Reply:
> However, the logic we are talking about here also impacts imperative update of the field, to enable users to test there application with fireEvents without caring about the implementation detail of the field, just by firing an onChange with the new value.

The `onChange` first checks if the modified value impacts only the selected section and if yes it only parses this section. So this should not be a problem.

If the modification goes outside of the selected section, then we will have the full parsing which can be problematic.

The only solution I see is to build a regexp (or a custom function) to do the cleaning

```js
const cleanValue = (inputValue, sections) => {
	let cleanedValue = inputValue
	sections.forEach(({startSeparator, separator}) => {
		cleanedValue = cleanedValue.replace(startSeparator, ' ')
		cleanedValue = cleanedValue.replace(separator, ' ')
	})
	// remove duplicate spaces
	// ...
	return cleanedValue
}
```

Reply:
In the `onChange` I have an early check saying that, if no section is selected, then it's probably an imperative update, in which case I re-parse the whole string just like for pasting (in its current form).

The selection will often go outside of the current form, because I don't expect people to respect all the RTL / hidden spaces characters.
But the current approach works fine, if no section is selected, then parse the string based on the format.
I thought it was more complicated than that because I was trying to parse `[The] 6th [of] January`, but parsing `The 6th of January` works great.